### PR TITLE
A couple improvements

### PR DESCRIPTION
**Pull request by Lennix**
Created RpcClient.Startup which implements client-like behavior

Moved start of heartbeatDispatcher to StartDispatcher to start it after Startup is complete (and EventHandlers subscribed)

Use _session.GlobalSettingHash for DownloadSettingMessage requests

### CODE REVIEW

File: POGOLib/Net/Session.cs [hunk 1]
```diff
@@ -31,12 +31,20 @@ internal Session(AccessToken accessToken, string password, GeoCoordinate geoCoor
             Map = new Map(this);
 
             RpcClient = new RpcClient(this);
-            RpcClient.SendRemoteProcedureCall(RequestType.GetPlayer);
-            RpcClient.RefreshMapObjects();
 
             _heartbeat = new HeartbeatDispatcher(this);
         }
 
+        public bool Startup()
```
Comment (AeonLucid) [inline POGOLib/Net/Session.cs:38]: What happens when startup is called after startup already happened?

File: POGOLib/Net/RPCClient.cs [hunk 2]
```diff
@@ -69,6 +69,68 @@ internal RpcClient(Session session)
         /// <summary>
```
Comment (AeonLucid) [inline POGOLib/Net/RPCClient.cs:69]: Wrong docs for this method.

File: POGOLib/Net/RPCClient.cs [hunk 3]
```diff
@@ -69,6 +69,68 @@ internal RpcClient(Session session)
...
         /// It is not recommended to call this. Map objects will update automatically and fire the <see cref="Map.Update"/> event.
         /// </summary>
+        public bool Startup()
```
Comment (AeonLucid) [inline POGOLib/Net/RPCClient.cs]: Change to internal as it should only be called from the session startup.

**Comment by AeonLucid** [comment]
Hey, looks pretty good. I've added some notes, do you think you could make those changes? Will be happy to merge after.

**Comment by AeonLucid** [comment]
Also, could you maybe update the demo to reflect your changes regarding `Startup()`? 

I am currently on holiday so I don't have access to stuff to code, just my phone. :(

**Comment by Lennix** [comment]
Sure, will do

File: POGOLib/Net/Session.cs (hunk 1 above)
Comment (Lennix) [inline POGOLib/Net/Session.cs:38]: It sends the requests again

**Comment by Lennix** [comment]
Merged via PullRequest #18

**Pull request closed without merging**